### ISSUE TITLE
[BUG] filter / select producing Masked instead of `Indexed`.

### ISSUE THREAD

**Issue by derifatives** [genjax-community/genjax#696]
```
s = genjax.select({'chain': genjax.indexed_select([3], 'position')})
tr.filter(s)

└── :chain
    └── (Mask,  bool[1])
        └── :position
            └──  f32[1]
```

This is from an `Unfold` trace.

**Comment by sritchie** [comment]
If I respond here, this will show up on GitHub.

**Comment by sritchie** [comment]
Response to sam.

**Comment by sritchie** [comment]
Closing this since it doesn't reproduce in the new codebase.
